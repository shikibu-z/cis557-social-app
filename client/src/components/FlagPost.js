/* eslint-disable react/jsx-props-no-spreading */
import {
  Modal, Form, Button,
} from 'react-bootstrap';
import { React } from 'react';
import '../stylesheets/Editprofile.css';

const lib = require('../api/fetchers');

function FlagPost(props) {
  const {
    show, onHide, handleFlag, handleClose, groupId, postId, userId,
  } = props;

  const reportDelete = () => {
    lib.hideOrFlagPost(groupId, postId, { flag: true, hide: false, userId });
    handleFlag();
  };

  return (
    <Modal show={show} onHide={onHide}>
      <Modal.Header closeButton />
      <Modal.Body>
        <div>
          <h4>Flag this post for deletion?</h4>
          <Form>
            <Form.Group className="mb-3" id="deactivateButton">
              <Button
                onClick={handleClose}
                style={{
                  backgroundColor: 'red', color: 'white', position: 'absolute', left: '35%',
                }}
              >
                No
              </Button>
            </Form.Group>
            <Button onClick={reportDelete} variant="primary">
              Yes
            </Button>
          </Form>
        </div>
      </Modal.Body>
    </Modal>
  );
}

export default FlagPost;
