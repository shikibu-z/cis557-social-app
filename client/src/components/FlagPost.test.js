/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import FlagPost from './FlagPost';

test('Flag post matches snapshot', () => {
  const component = renderer.create(<FlagPost />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
