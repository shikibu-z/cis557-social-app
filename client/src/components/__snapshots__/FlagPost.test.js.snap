// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag post matches snapshot 1`] = `null`;
